src/App.tsx

```diff
@@ -1,54 +1,94 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState<Person | null>(null);
+  const [isInputOnFocus, setIsInputOnFocus] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const delay = useRef<number>(1000);
+
+  const applyQuery = useCallback(
+    debounce(setQuery, delay.current),
```

Comment:
Why do you create ref for the delay?
```suggestion
debounce(setQuery, 1000),
```